### DISCUSSION TITLE
Track who is sending a query from explore

### DISCUSSION THREAD

**Discussion by tanner-bruce** [grafana/clickhouse-datasource#690] [Ideas]
It would be helpful to be able to track who ran a query from explore, e.g if inefficient queries are identified we could use this to reach out and help users writer better queries

If I check the stats in system.query_log this is what grafana is sending. `forwarded_for` is just the IP of grafana itself. Maybe the user could be included in either user agent, or as client_name?

```
client_hostname:
client_name:
client_revision:                       0
client_version_major:                  0
client_version_minor:                  0
client_version_patch:                  0
http_method:                           2
http_user_agent:                       grafana/0.0.0 clickhouse-datasource/4.0.0-beta clickhouse-go/2.16.0 (lv:go/1.21.5; os:linux)
http_referer:
forwarded_for:                         107...
```

**Comment by andaag** [comment]
We have redash running as well as grafana, and redash injects the current users information in the log_comment field as a parsable json. This is *very* useful, as we can attribute all queries back to source dashboard and or user.

**Comment by SpencerTorres** [comment]
As of 4.0.3, we have this feature available:

https://github.com/grafana/clickhouse-datasource/discussions/667 (discussion)
https://github.com/grafana/clickhouse-datasource/pull/670 (pull request)

Would this feature help?

I see you want the user as part of the `client_name`. Right now this goes based off of the datasource and not per user, so we couldn't add it per query. I recall seeing some settings in the plugin SDK that creates a new connection per user, but that would use up a lot of connections depending on the deployment/usage. Let me know if we should explore that further, or if the existing features solve your issue.

Thanks!
